src/Kiota.Builder/Refiners/GoRefiner.cs

```diff
@@ -110,62 +107,30 @@ public override void Refine(CodeNamespace generatedCode)
-                            CodeParameterKind.ResponseHandler => CodePropertyKind.ResponseHandler,
-                            _ => CodePropertyKind.Custom
-                        },
-                    }).ToArray());
-                parameterSetClass.Properties.ToList().ForEach(x => {x.Type.ActionOf = false;});
-                currentMethod.RemoveParametersByKind(CodeParameterKind.RequestBody,
-                                                    CodeParameterKind.QueryParameter,
-                                                    CodeParameterKind.Headers,
-                                                    CodeParameterKind.Options,
-                                                    CodeParameterKind.ResponseHandler);
-                var parameterSetParameter = new CodeParameter{
-                    Name = "options",
-                    Type = new CodeType {
-                        Name = parameterSetClass.Name,
-                        ActionOf = false,
-                        IsNullable = true,
-                        TypeDefinition = parameterSetClass,
-                        IsExternal = false,
-                    },
-                    Optional = false,
-                    Description = "Options for the request",
-                    Kind = CodeParameterKind.ParameterSet,
-                };
-                currentMethod.AddParameter(parameterSetParameter);
-                var generatorMethod = parentClass.GetMethodsOffKind(CodeMethodKind.RequestGenerator)
-                                                .FirstOrDefault(x => x.HttpMethod == currentMethod.HttpMethod);
-                if(!(generatorMethod?.Parameters.Any(x => x.IsOfKind(CodeParameterKind.ParameterSet)) ?? true)) {
-                    generatorMethod.RemoveParametersByKind(CodeParameterKind.RequestBody,
-                                                    CodeParameterKind.QueryParameter,
-                                                    CodeParameterKind.Headers,
-                                                    CodeParameterKind.Options);
-                    generatorMethod.AddParameter(parameterSetParameter);
-                }
+    private void InsertOverrideMethodForRequestExecutorsAndBuildersAndConstructors(CodeElement currentElement) {
+        if(currentElement is CodeClass currentClass) {
+            var codeMethods = currentClass.Methods;
+            if(codeMethods.Any(x => x.IsOfKind(CodeMethodKind.RequestExecutor, CodeMethodKind.RequestGenerator))) {
+                var originalExecutorMethods = codeMethods.Where(x => x.IsOfKind(CodeMethodKind.RequestExecutor)).ToList();
+                var executorMethodsToAdd = originalExecutorMethods
+                                    .Select(x => GetMethodClone(x, CodeParameterKind.ResponseHandler, CodeParameterKind.ResponseHandler))
+                                    .Where(x => x != null)
+                                    .ToArray();
```

Comment:
please keep them as `IEnumerable` until the array type is needed. All the intermediate operations below (i.e. `Any()` and `Union()`) are supported on `IEnumerable`s as well.
```suggestion
                                    .Where(x => x != null);
```